src/IO/HTTPCommon.cpp

```diff
@@ -74,8 +74,12 @@ namespace
         if (https)
         {
 #if USE_SSL
-            /// Cannot resolve host in advance, otherwise SNI won't work in Poco.
-            session = std::make_shared<Poco::Net::HTTPSClientSession>(host, port);
+            String resolved_host = resolve_host ? DNSResolver::instance().resolveHost(host).toString() : host;
+            auto https_session = std::make_shared<Poco::Net::HTTPSClientSession>(host, port);
+            if (resolve_host)
+                https_session->setResolvedHost(DNSResolver::instance().resolveHost(host).toString());
```

Comment:
```suggestion
                https_session->setResolvedHost(resolved_host);
```

Reply:
There is also a poco patch which adds `setResolvedHost` method.
At least, `SELECT count() FROM url('https://data.gharchive.org/2015-01-01-15.json.gz', TSV, 'data String')` still works :)